x/gamm/keeper/grpc_query.go

```diff
@@ -184,14 +175,14 @@ func (q Querier) PoolsWithFilter(ctx context.Context, req *types.QueryPoolsWithF
 		checks_needed++
 	}
 
-	for _, p := range pools {
+	var response = []*codectypes.Any{}
+	pageRes, err := query.FilteredPaginate(poolStore, req.Pagination, func(_, value []byte, accumulate bool) (bool, error) {
 		var checks = 0
-		var pool types.PoolI
-
-		err := q.cdc.UnpackAny(p, &pool)
+		pool, err := q.Keeper.UnmarshalPool(value)
```

Comment:
where does this `value` as the input to UnmarshalPool come from?

Reply:
It comes from the FilteredPagination method, where it iterates over the given kv store!